### PR TITLE
Verify task writer count and task concurrency are a power of two

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by electrum** [comment]
Where is the error message for `@PowerOfTwo`? What does the user see if it fails?

**Comment by dain** [comment]
I copied the `@MinDuration` code @martint wrote.  There doesn't seem to be a message for that either.

**Comment by electrum** [comment]
I'd like to see that the useyou gets an understandable error when validation fails. Otherwise looks good.

File: presto-main/src/main/java/com/facebook/presto/util/PowerOfTwoValidator.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+    @Override
+    public void initialize(PowerOfTwo powerOfTwo)
+    {
+    }
+
+    @Override
+    public boolean isValid(Integer value, ConstraintValidatorContext context)
+    {
+        return value == null || value == 0 || Integer.bitCount(value) == 1;
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/util/PowerOfTwoValidator.java]: 0 is not a power of 2.

**Comment by electrum** [comment]
Looks good

**Pull request closed without merging**